### PR TITLE
GRAPH.INFO QUERIES subcommand

### PR DESCRIPTION
**Pull request by raz-mon**
This PR implements the `GRAPH.INFO QUERIES [CURRENT] [PREV<count>]` subcommand of `GRAPH.INFO`.

### CODE REVIEW

File: src/configuration/config.h [hunk 1]
```diff
@@ -30,6 +30,8 @@ typedef enum {
 	Config_QUERY_MEM_CAPACITY        = 10,  // max mem(bytes) that query/thread can utilize at any given time
 	Config_DELTA_MAX_PENDING_CHANGES = 11,  // number of pending changes before RG_Matrix flushed
 	Config_NODE_CREATION_BUFFER      = 12,  // size of buffer to maintain as margin in matrices
+	Config_CMD_INFO                  = 13,  // toggle on/off the GRAPH.INFO
+	Config_CMD_INFO_MAX_QUERY_COUNT  = 14,  // the max number of info queries count
```
Comment (LiorKogan) [inline src/configuration/config.h:34]: > Config_END_MARKER. Intentional?

Comment (raz-mon) [reply src/configuration/config.h:34]: Should be 15 of course, thanks.

File: docs/docs/configuration.md [hunk 2]
```diff
@@ -285,6 +285,22 @@ This configuration can be set when the module loads or at runtime.
...
+
+An on/off toggle for the `GRAPH.INFO` command. Disabling this command may increase performance and lower the memory usage and these are the main reasons for it to be disabled.
+
+#### Default
+
+`CMD_INFO` is `yes`.
+
+### MAX_INFO_QUERIES
+
+A limit for the number of finished queries stored for the `GRAPH.INFO PREV` subcommand.
```
Comment (LiorKogan) [inline docs/docs/configuration.md:298]: Please change `finished queries` to `previously-executed queries`

Comment (LiorKogan) [reply docs/docs/configuration.md:298]: Please also add limits (0 - 10000?)

File: docs/docs/configuration.md [hunk 3]
```diff
@@ -285,6 +285,22 @@ This configuration can be set when the module loads or at runtime.
 
 `VKEY_MAX_ENTITY_COUNT` is 100,000.
 
+### CMD_INFO
+
+An on/off toggle for the `GRAPH.INFO` command. Disabling this command may increase performance and lower the memory usage and these are the main reasons for it to be disabled.
+
+#### Default
+
+`CMD_INFO` is `yes`.
```
Comment (LiorKogan) [inline docs/docs/configuration.md:294]: Please list the valid values: e.g., 'yes' ..., 'no' ...

File: docs/docs/configuration.md [hunk 4]
```diff
@@ -285,6 +285,22 @@ This configuration can be set when the module loads or at runtime.
...
+
+### MAX_INFO_QUERIES
+
+A limit for the number of finished queries stored for the `GRAPH.INFO PREV` subcommand.
+
+#### Default
+
+`MAX_INFO_QUERIES` is 10,000.
```
Comment (LiorKogan) [inline docs/docs/configuration.md:302]: "10,000" is misleading, I think "10000" is better (someone may set it to "5,000").

File: docs/docs/configuration.md (hunk 2 above)
Comment (raz-mon) [reply docs/docs/configuration.md:298]: We did not define such a maximum value for the number of queries we can store. We assigned a uint_32 for it, but I'm not sure we want to allow that much queries to be saved.

Comment (LiorKogan) [reply docs/docs/configuration.md:298]: I think 10000. I actually mentioned "If larger than 10000 it is replaced with 10000"

Comment (raz-mon) [reply docs/docs/configuration.md]: Why 10000?
Not that I have a specific problem with that number (nice and round), but I think we should think of some limit on the memory or an amount of 'relevant' queries we see right.

Comment (LiorKogan) [reply docs/docs/configuration.md]: We can decide based on the maximal memory size we should allow for this. 20MB? I don't know how much each entry takes. Another consideration is assuming an average throughput of e.g., 250 queries/sec for saving e.g., 60 seconds (which is 15000 in this case). @swilly22 @AviAvni what is your take here?

**Comment by filipecosta90** [comment]
### Automated performance analysis summary

This comment was automatically generated given there is performance data available.

In summary:
- Detected a total of 10 stable tests between versions.

You can check a comparison in detail via the [grafana link](https://benchmarksrediscom.grafana.net/d//SH9_rQYGz/?var-branch=master&var-branch=graph-info-queries)

### Comparison between master and graph-info-queries.

Time Period from 30 days ago. (environment used: oss-standalone)

|                         Test Case                          |Baseline master (median obs. +- std.dev)|Comparison graph-info-queries (median obs. +- std.dev)|% change (higher-better)|     Note      |
|------------------------------------------------------------|----------------------------------------|------------------------------------------------------|------------------------|---------------|
|ENTITY_COUNT                                                | 49998 +- 0.0%  (3 datapoints)          | 49998 +- nan%  (1 datapoints)                        |-0.0%                   |-- no change --|
|EXPLICIT-EDGE-DELETION                                      | 100 +- 0.0%  (3 datapoints)            | 100 +- nan%  (1 datapoints)                          |0.0%                    |-- no change --|
|GRAPH500-SCALE_18-EF_16-1_HOP                               | 33324 +- 0.0%  (3 datapoints)          | 33323 +- nan%  (1 datapoints)                        |-0.0%                   |-- no change --|
|GRAPH500-SCALE_18-EF_16-2_HOP_MIXED_READ_75perc_WRITE_25perc| 59 +- 3.2%  (3 datapoints)             | 59 +- nan%  (1 datapoints)                           |0.0%                    |-- no change --|
|IMPLICIT-EDGE-DELETION                                      | 17 +- 0.0%  (3 datapoints)             | 17 +- nan%  (1 datapoints)                           |-0.0%                   |-- no change --|
|INSPECT-EDGES                                               | 2500 +- 0.0%  (3 datapoints)           | 2500 +- nan%  (1 datapoints)                         |-0.0%                   |-- no change --|
|MIX_READ_WRITE                                              | 10000 +- 0.0%  (3 datapoints)          | 10000 +- nan%  (1 datapoints)                        |-0.0%                   |-- no change --|
|NODE-INDEX-LOOKUP                                           | 33 +- 0.0%  (3 datapoints)             | 33 +- nan%  (1 datapoints)                           |-0.0%                   |-- no change --|
|SORT_ENTITIES                                               | 100 +- 0.0%  (3 datapoints)            | 100 +- nan%  (1 datapoints)                          |-0.0%                   |-- no change --|
|allShortestPaths-10hop-100Kpaths                            | 30 +- 0.0%  (3 datapoints)             | 30 +- nan%  (1 datapoints)                           |0.0%                    |-- no change --|

**Pull request merged**